.github/workflows/publish.yml

```diff
@@ -34,13 +36,16 @@ jobs:
 
       - name: ❓ Check if a new version needs to be published
         id: publish-check
-        continue-on-error: true
-        run: yarn publish:check
+        run: echo "SHOULD_PUBLISH=$(yarn publish:check)" >> "$GITHUB_OUTPUT"
+
+      - name: ⏹️ Cancel run
```

Comment:
This step is technically not required, as if the `SHOULD_PUBLISH` is `yes`, then only the other two steps will be triggered.

But calling `gh run cancel` will mark the workflow as cancelled instead of succeeding, which is more readable in the UI
![image](https://github.com/algolia/npm-search/assets/283419/0008307f-10ec-46ab-9cb3-6eed800aa21f)